app/models/mdm/payload.rb

```diff
@@ -0,0 +1,118 @@
+# A payload that has been generated by Metasploit and used to establish {Mdm::Session}.
+class Mdm::Payload < ActiveRecord::Base
+  extend ActiveSupport::Autoload
+
+  include Metasploit::Model::Search
+
+  #
+  # Associations
+  #
+
+  # @!attribute [rw] workspace
+  # {Mdm::Workspace} in which this payload was created.
+  #
+  #   @return [Mdm::Workspace]
+  belongs_to :workspace,
+             class_name: 'Mdm::Workspace',
+             inverse_of: :payloads
+
+
+  #
+  # Attributes
+  #
+
+  # @!attribute [rw] name
+  #   The name of this payload.
+  #
+  #   @return [String]
+
+  # @!attribute [rw] uuid
+  #   A unique identifier for this payload. The UUID is encoded to include specific information.
+  #   See lib/msf/core/payload/uuid.rb in the https://github.com/rapid7/metasploit-framework repo.
+  #
+  #   @return [String]
+
+  # @!attribute [rw] uuid_mask
+  #   The number of bytes of the UUID that this payload has embedded within it. This is to support
+  #   legacy payloads that limit the UUID to 8 bytes
+  #
+  #   @return [Integer]
+
+  # @!attribute [rw] timestamp
+  #   The Unix format timestamp when this payload was created.
+  #
+  #   @return [Integer]
+
+  # @!attribute [rw] arch
+  #   The architecture this payload supports.
+  #   Valid values are located at lib/msf/core/payload/uuid.rb in the https://github.com/rapid7/metasploit-framework repo.
+  #
+  #   @return [String]
+
+  # @!attribute [rw] platform
+  #   The platform this payload supports.
+  #   Valid values are located at lib/msf/core/payload/uuid.rb in the https://github.com/rapid7/metasploit-framework repo.
+  #
+  #   @return [String]
+
+  # @!attribute [rw] urls
+  #   The unique, encoded urls used to host this payload. Only applicable for http(s) payloads
+  #
```

Comment:
Maybe this should get shuffled off as a secondary table?  @bcook-r7 and I were taking that this would be handy as a generic secondary-attribute table to support multiple URI or secondary-identifier attributes for a listener like email address, twitter handle, or something we have not yet figured out?  Splitting it to a second table would allow multiple secondary attributes on the same payload.  For instance, maybe we'd like to have a payload cycle through multiple URIs in callback so it does not repeat itself?

Reply:
The use case for the secondary attribute is a handler should be able to get a connection, look it up, and identify the payload UUID that matches it. It could be a many-to-one mapping.